### PR TITLE
Windows buttons

### LINKED ISSUE #1: support windows and other platforms
awesome that you pulled this out! we should add windows/ubuntu styling to this as well you can use this to get a platform specific titlebar

### PR DESCRIPTION
**Pull request by reeseovine**
Adds option for styling title bar like Windows 10. Set an option named 'os' to 'mac' or 'win'.

Also fixes #1.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
   "name": "titlebar",
-  "version": "1.4.0",
+  "version": "1.5.0",
   "description": "Emulate OS X window title bar",
   "main": "index.js",
   "scripts": {
-    "demo-watch": "wzrd index.js:demo/index.js -- -s titlebar",
-    "demo-build": "browserify index.js -o demo/index.js -s titlebar"
+    "start": "wzrd index.js:demo/index.js -- -s titlebar",
+    "test": "browserify -t brfs index.js -o demo/index.js -s titlebar"
```
Comment (kapetan) [inline package.json]: Is it necessary to rename these? I don't think `-t brfs` is need, the package.json already contains the browserify transform field.

File: index.js [hunk 2]
```diff
@@ -48,6 +76,9 @@ var TitleBar = function(options) {
 util.inherits(TitleBar, events.EventEmitter);
 
 TitleBar.prototype.appendTo = function(target) {
+	if (this._options.os == 'mac') var style = styleMac;
+	else if (this._options.os == 'win') var style = styleWin;
```
Comment (kapetan) [inline index.js]: Looks kind of weird. It would be better to declare `var style` outside the if statement.

Comment (kapetan) [inline index.js]: To keep with rest of the code style use `===` for comparison.

File: index.js [hunk 3]
```diff
@@ -17,6 +19,16 @@ var TitleBar = function(options) {
...
 	this._options = options || {};
 
+	if (this._options.os == 'mac' && this._options.os == 'win'){
+		if (process.platform == 'darwin') this._options.os = 'mac';
+		else if (process.platform == 'win32') this._options.os = 'win';
+		else {
+			console.error('No supported os style type was given. Using OS X style as default.')
+			this._options.os = 'mac';
+		}
+	}
```
Comment (kapetan) [inline index.js]: The whole if statement is never executed, the `os` option can't be `mac` and `win` at the same time.

File: index.js [hunk 4]
```diff
@@ -17,6 +19,16 @@ var TitleBar = function(options) {
...
+	if (this._options.os == 'mac') var html = htmlMac;
+	else if (this._options.os == 'win') var html = htmlWin;
```
Comment (kapetan) [inline index.js]: The `os` option should be optional, at the moment it throws an error when not specified.

File: demo/index.html [hunk 5]
```diff
@@ -12,12 +12,20 @@
...
 			.on('fullscreen', log('fullscreen'))
 			.on('maximize', log('maximize'));
+
+		document.write('<br>');
+
+		var tw = titlebar({'os':'win'})
+			.appendTo(document.body)
+			.on('close', log('close'))
+			.on('minimize', log('minimize'))
+			.on('maximize', log('maximize'));
```
Comment (kapetan) [inline demo/index.html:28]: I've noticed that the titlebar changes height when the maximize button is pressed.

File: index.js [hunk 6]
```diff
@@ -26,15 +38,31 @@ var TitleBar = function(options) {
...
+		else if (os == 'mac'){
+			if(fullscreen.contains(target)) {
+				if(e.altKey) self.emit('maximize', e);
+				else self.emit('fullscreen', e);
+			}
+		}
+		else if (os == 'win'){
+			if(maximize.contains(target)) {
+				if($element.hasClass('maximized')){
+					self.emit('unmaximize', e);
```
Comment (kapetan) [inline index.js]: This should be `minimize`.

**Comment by kapetan** [comment]
Thanks for the pull-request. I left a couple of comments, you maybe could address.

File: package.json (hunk 1 above)
Comment (reeseovine) [inline package.json]: Oh I guess not. That was left over from my own testing. You can revert that.

File: index.js (hunk 2 above)
Comment (reeseovine) [inline index.js]: I was having some trouble figuring out where to put that check. It's a little convoluted with the way the style and dom are appended.

File: index.js (hunk 3 above)
Comment (reeseovine) [inline index.js]: Oh shoot, I meant to check if the option was neither 'mac' nor 'win', but I must not have tested that. I can redo that part if you need.

File: index.js (hunk 4 above)
Comment (reeseovine) [inline index.js]: Same as above.

File: demo/index.html (hunk 5 above)
Comment (reeseovine) [inline demo/index.html:28]: That's intentional. It's the behavior Windows 10 uses when maximizing, and I checked that the height is correct.

File: index.js (hunk 6 above)
Comment (reeseovine) [inline index.js]: You mean the 'unmaximize' event? That and minimize are not the same.

**Comment by reeseovine** [comment]
Hey, I finally got around to fixing those issues you helped me find. Please review my new edits. Thanks!

**Comment by ajbeaven** [comment]
Thanks for this pull request @katacarbix. Would be great to see it merged!

**Comment by EricRabil** [comment]
Seeing as this is unlikely to be merged any time soon, would you mind publishing your fork to the NPM repository?

**Comment by reeseovine** [comment]
Sure! I should tidy it up a bit, but that sounds like a great idea.

**Comment by ghost** [comment]
oh let me do that, i am sure i can make it ~~3~~ less than 30 lines with modern javascript

**Comment by reeseovine** [comment]
Decided to split this off into my own project, which I'm hoping will be an all-in-one solution for titlebars. Check it out here: https://github.com/katacarbix/windowbar

**Pull request closed without merging**